app/src/main/java/io/appium/uiautomator2/model/UiElementSnapshot.java

```diff
@@ -277,6 +278,24 @@ private void addToastMsg(CharSequence tokenMSG) {
                 new HashSet<>(Arrays.asList(TOAST_NODE_ATTRIBUTES))));
     }
 
+    /**
+     * Detects if the tree traversal should continue, because we hope to find visible children under
+     * this node
+     * @param className the class name of the current node
+     * @param traversableClassPrefixes a list of class prefixes that
+     * @see <a href="https://issuetracker.google.com/issues/354958193">https://issuetracker.google.com/issues/354958193</a>
+     * @return true, if the given node should be traversed further else false
+     */
+    private boolean mayContainVisibleChildren(String className, String[] traversableClassPrefixes) {
```

Comment:
The `className` can be null so please add `nullable` annotation and add null case handling for `className`